service/src/main/java/bio/terra/drshub/services/DrsResolutionService.java

```diff
@@ -235,6 +249,7 @@ DrsObject fetchObjectInfo(
     log.info(drsRequestLogMessage);
 
     var drsApi = drsApiFactory.getApiFromUriComponents(uriComponents, drsProvider);
+    drsApi.setHeader("x-transaction-id", transactionId);
```

Comment:
`X-Transaction-ID` could probably be extracted into a variable since its referenced multiple times.

Reply:
I don't know if we want to add the transaction id header to every request. Other DRS Providers might have similar header restrictions to us, and not allow it through. We should probably only add it if we know the request is going to TDR.

I think I might have said the opposite when we talked about it face to face, but I'm realizing now that if we're restricting what headers are allowed in requests to our systems, others are probably doing it too.

Reply:
We're not checking the provider for other headers btw 😅 At least we always pass the IP address and x-user-project for access url requests.

Reply:
Oh, maybe it's fine then!